src/eth_provider/database/ethereum.rs

```diff
@@ -29,12 +39,33 @@ pub trait EthereumDatabase {
 }
 
 #[async_trait]
-impl EthereumDatabase for Database {
+impl EthereumTransactionStore for Database {
     async fn transaction(&self, hash: &B256) -> Result<Option<Transaction>, EthApiError> {
         let filter = EthDatabaseFilterBuilder::<filter::Transaction>::default().with_tx_hash(hash).build();
         Ok(self.get_one::<StoredTransaction>(filter, None).await?.map(Into::into))
     }
 
+    async fn transactions(&self, hash_or_number: BlockHashOrNumber) -> Result<Vec<Transaction>, EthApiError> {
```

Comment:
Here I don't really have a better solution for the name of the variable `hash_or_number` but it gives me the impression that the naming is not complete because it could be confused with a transaction hash or number.
- On the reth codebase side, I have often seen the name `block` for this type but it is not really explicit.
- Maybe we could put `block_hash_or_number` but it gets a little long...

Reply:
personnally ok with `block_hash_or_number`

Reply:
Let's go with this then. I think there are other identical variables in this PR to be modified too.